Darktrace/darktrace_modules/threat_visualizer_log_trigger.py

```diff
@@ -0,0 +1,131 @@
+import time
+import traceback
+from functools import cached_property
+from urllib.parse import urljoin
+
+import orjson
+import requests
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.storage import PersistentJSON
+
+from darktrace_modules import DarktraceModule
+from darktrace_modules.client import ApiClient
+from darktrace_modules.logging import get_logger
+
+logger = get_logger()
+
+
+class ThreatVisualizerLogConnectorConfiguration(DefaultConnectorConfiguration):
+    frequency: int = 1
+    ratelimit_per_minute: int = 83
+    filter: str | None = None
+    q: str | None = None
+
+
+class ThreatVisualizerLogConnector(Connector):
+    """
+    This connector fetches audit logs from the Darktrace API
+    """
+
+    module: DarktraceModule
+    configuration: ThreatVisualizerLogConnectorConfiguration
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+
+    @property
+    def last_ts(self) -> int:
+        with self.context as cache:
+            ts = cache.get("last_ts")
+            if ts is None:
+                ts = round(time.time() * 1000)
+                self.last_ts = ts
+            return ts
+
+    @last_ts.setter
+    def last_ts(self, last_ts: str):
+        with self.context as cache:
+            cache["last_ts"] = last_ts
+
+    def stop(self, _, __):
+        self.log(message="Stopping Darktrace threat visualizer logs connector", level="info")
+        # Exit signal received, asking the processor to stop
+        super().stop()
+
+    @cached_property
+    def client(self):
+        return ApiClient(
+            self.module.configuration.public_key,
+            self.module.configuration.private_key,
+            ratelimit_per_minute=self.configuration.ratelimit_per_minute,
+        )
+
+    def request_events(self) -> requests.models.Response:
+        params = {"starttime": str(self.last_ts), "includeallpinned": "false"}
+        url = urljoin(self.module.configuration.api_url, "modelbreaches")
+
+        response = self.client.get(url, params=params)
+        return response
+
+    def refine_response(self, response: list) -> list:
+        # as we use the time variable of the newest event to set last_ts, this event has to be removed in the next batch of events.
+        if response[0]["time"] == self.last_ts:
+            return response[1:]
+        return response
+
+    def next_batch(self):
+        # save the start time
+        batch_start_time = time.time()
+        response = []
+        try:
+            response = self.request_events().json()
```

Comment:
I do think all events are returned in one page. I searched "pagination" in their documentation and didn't find anything. Even if they have pages when they reach a certain number of events (which I doubt), the connector will still work as it will ask the events from the newest he got (they are sent from oldest to newest).